src/Stratis.SmartContracts.Executor.Reflection/Serialization/MethodParameterStringSerializer.cs

```diff
@@ -118,49 +118,58 @@ public object[] Deserialize(string parameters)
+                        processedParameters.Add(uint.Parse(parameterSignature[1]));
 
-                else if (parameterSignature[0] == MethodParameterDataType.String.ToString("d"))
-                    processedParameters.Add(parameterSignature[1]);
-                
-                else if (parameterSignature[0] == MethodParameterDataType.UInt.ToString("d"))
-                    processedParameters.Add(uint.Parse(parameterSignature[1]));
+                    else if (parameterSignature[0] == MethodParameterDataType.Int.ToString("d"))
+                        processedParameters.Add(int.Parse(parameterSignature[1]));
 
-                else if (parameterSignature[0] == MethodParameterDataType.Int.ToString("d"))
-                    processedParameters.Add(int.Parse(parameterSignature[1]));
+                    else if (parameterSignature[0] == MethodParameterDataType.ULong.ToString("d"))
+                        processedParameters.Add(ulong.Parse(parameterSignature[1]));
 
-                else if (parameterSignature[0] == MethodParameterDataType.ULong.ToString("d"))
-                    processedParameters.Add(ulong.Parse(parameterSignature[1]));
+                    else if (parameterSignature[0] == MethodParameterDataType.Long.ToString("d"))
+                        processedParameters.Add(long.Parse(parameterSignature[1]));
 
-                else if (parameterSignature[0] == MethodParameterDataType.Long.ToString("d"))
-                    processedParameters.Add(long.Parse(parameterSignature[1]));
-                
-               else if (parameterSignature[0] == MethodParameterDataType.Address.ToString("d"))
-                    processedParameters.Add(parameterSignature[1].ToAddress(this.network));
+                    else if (parameterSignature[0] == MethodParameterDataType.Address.ToString("d"))
+                        processedParameters.Add(parameterSignature[1].ToAddress(this.network));
 
-                else if (parameterSignature[0] == MethodParameterDataType.ByteArray.ToString("d"))
-                    processedParameters.Add(parameterSignature[1].HexToByteArray());
+                    else if (parameterSignature[0] == MethodParameterDataType.ByteArray.ToString("d"))
+                        processedParameters.Add(parameterSignature[1].HexToByteArray());
 
-                else
-                    throw new Exception(string.Format("{0} is not supported.", parameterSignature[0]));
+                    else
+                        throw new MethodParameterStringSerializerException(string.Format("{0} is not supported.",
+                            parameterSignature[0]));
+                }
+            }
+            catch (Exception e) when (e is FormatException || e is OverflowException || e is ArgumentException || e is ArgumentNullException)
+            {
+                throw new MethodParameterStringSerializerException(e.Message);
```

Comment:
How about we use the InnerException functionality and pass the exception into the constructor as well, to maintain any extra data on the exception object

Reply:
Good idea, will add.